docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+
+    [CPEMessageQueue]
+    Protocol = 'tcp'
+    Host = '*'
+    Port = 5574
+    Type = 'zero'
+    Topic = 'cpe'
+    [MessageQueue.Optional]
+        # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+        # Client Identifiers
+        Username =""
+        Password =""
+        ClientId ="service-abc"
+        # Connection information
+        Qos = "0" # Quality of service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive =  "10" # Seconds (must be 2 or greater)
+        Retained = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+        SkipCertVerify = "false"
+  ```
+
+- Metrics and CPE collected/monitored and reported by a service will be associated with a name key.  That name key will be used in the configuration to turn `on` the metric or CPE and will be used in the model/DTO when reporting on the metric or CPE. 
+- *Should consideration be given to incorporate alternate protocols/standards for metric and CPE such as https://opentelemetry.io/ or https://github.com/statsd/?*
+- *Should we provide a standard interface for the function that gets called to create a new CPE*
+- *Should we provide a standard interface for the function that gets called by a schedule service (internal or external) to fetch and publish the desired metrics at the appointed time?*
+  - *Do we dictate the use of the scheduler service for this or use the internal scheduler approach?*
+
+## Decision
+
+*To be determined*
+
+## Consequences
+
+- Given the potential that each service publish metrics or CPE, 0MQ is an unlikely implementation unless each service uses a different 0MQ pipe (0MQ topics do not allow multiple publishers).  *Given this constraint, should 0MQ implementation be allowed*
+- We need to avoid service bloat.  EdgeX is not an enterprise system.  How can we implement in a concise and economical way?
+- The existing system management service (and associated executors) shall remain in place for now.  SMA reports on service CPU, memory, configuration and provides the means to start/stop/restart the services.  This is currently outside the scope of the new metric/CPE collection/monitoring.
+  - In the future, SMA could be configured to subscribe to the metric/CPE messages and provide those to clients (with REST calls in a pull vs push way).

```

Comment:
We need to keep the bloat down and only have push on message bus. SMQ can subscribe if  it needs the data.